### ISSUE TITLE
How to add an external plugin?

### ISSUE THREAD

**Issue by win32nipuh** [rawkode/bring-your-own-telegraf#4]
I'd like to build Docker Telegraf  with an external output plugin: postgresql.
The InfluxData does not contain this plugin but it exists here
https://github.com/svenklemm/telegraf

What I need to do build Talagraf+postgresql ouput plugin?
 
Thank you!

**Comment by rawkode** [comment]
@win32nipuh Hmm. I can add extra hooks for this. Can you let me know if this is still a use-case you wish to explore and I'll get something working for you.

**Comment by mengyu-dev** [comment]
yes, this is also a real use-case for our team in production. We develop a custom plugin and we want to use your super byot solution to optimize the docker image. but  we can't esaily integrate our custom plugin in the dockerfile. The hook should be useful.

**Comment by rawkode** [comment]
@mengyu-dev Do you keep your plugins in their own repository and you want them compiled against the Telegraf binary?

If so, I will work on this tomorrow.
